ngraph/core/include/ngraph/descriptor/tensor.hpp

```diff
@@ -58,13 +53,6 @@ namespace ngraph
             const element::Type& get_element_type() const { return m_element_type; }
             const Shape& get_shape() const;
             const PartialShape& get_partial_shape() const { return m_partial_shape; }
-            const std::shared_ptr<layout::TensorLayout>& get_tensor_layout() const
-            {
-                return m_tensor_layout;
-            }
-
-            void set_tensor_layout(const std::shared_ptr<layout::TensorLayout>& tensor_layout);
-
```

Comment:
@geexie, are you OK with this change?

Reply:
@slyalin, thanks for pinging me. I'm OK with this change, since I don't use this function and `TensorLayout` property. I use `AxisVector` for describing blocking. I don't feel like having something special about layout is generic enough solution so I came up with `AxisVector` idea to pin related dimensions. For example, for NCHW8c i have 5d `Shape` <N, C, H, W, 8> and 5d `AxisVector` <0, 1, 2, 3, 1>. Honestly, I'd love to integrate this approach into Tensor and account it in shape merging routine. What do you think about such an approach?

Reply:
Looks like you use the same approach like we use in the TensorIterator. Maybe we can think about selection of common structures, and use TensorDesc not only in the IE.

Reply:
👌  makes sense. let's discuss this in separate thread then?